argus/recipes/base.py

```diff
@@ -50,19 +57,35 @@ def __init__(self, instance_id, api_manager, remote_client, image,
         self._service_type = service_type
         self._output_directory = output_directory
 
-    def _execute(self, cmd):
-        """Execute and return only the standard output."""
+    def _execute(self, cmd, boot=False):
```

Comment:
I don't like that this function has implicit knowledge of where it should be used. Maybe you can specify the behaviour for these methods explicitly?